openedx/core/djangoapps/course_roles/helpers.py

```diff
@@ -0,0 +1,155 @@
+"""
+Helpers for the course roles djangoapp which is used for authorization.
+"""
+from django.contrib.auth.models import AnonymousUser, User  # lint-amnesty, pylint: disable=imported-auth-user
+from django.utils.translation import gettext as _
+
+from edx_toggles.toggles import WaffleFlag
+from openedx.core.djangoapps.course_roles.models import UserRole
+from xmodule.modulestore.django import modulestore
+
+
+# .. toggle_name: FLAG_USE_PERMISSION_CHECKS
+# .. toggle_implementation: WaffleFlag
+# .. toggle_default: False
+# .. toggle_description: Enabling the toggle will allow the db checks for a users permissions. These are used alongside current
+#   roles checks. If the flag is not enabled, only the roles checks will be used.
+# .. toggle_use_cases: temporary
+# .. toggle_creation_date: 2023-10-17
+# .. toggle_target_removal_date: 2023-12-01
+# .. toggle_warning:
+USE_PERMISSION_CHECKS_FLAG = WaffleFlag('course_roles.use_permission_checks', module_name=__name__) or False  # lint-amnesty, pylint: disable=toggle-missing-annotation
+
+
+def use_permission_checks():
+    """
+    Returns ture if permissions checks should be used
+    """
+    return USE_PERMISSION_CHECKS_FLAG.is_enabled()
+
+
+def user_has_permission_course(user, permission_name, course_id):
+    """
+    Check if a user has a permission in a course.
+    """
+    if not use_permission_checks():
+        return False
+    if isinstance(user, AnonymousUser) or not isinstance(user, User):
+        return False
+    return UserRole.objects.filter(
+        user=user,
+        role__permissions__name=permission_name,
+        course=course_id,
+    ).exists()
+
+
+def user_has_permission_list_course(user, permission_names, course_id):
+    """
+    Check if a user has ALL of the given permissions in a course.
+    """
+    if not use_permission_checks():
+        return False
+    return all(user_has_permission_course(user, permission_name, course_id) for permission_name in permission_names)
+
+
+def user_has_permission_list_any_course(user, permission_names, course_id):
```

Comment:
By convention, this param should be `course_key` instead of `course_id` if we're using a `CourseKey` type as the argument (and we probably should).